escrow/atomic_storage.go

```diff
@@ -33,6 +33,22 @@ type PrefixedAtomicStorage struct {
 	keyPrefix string
 }
 
+//It is recommended to use this function to create a PrefixedAtomicStorage
+func NewPrefixedAtomicStorage(atomicStorage AtomicStorage, prefix string) *PrefixedAtomicStorage {
+	if err, ok := atomicStorage.(*PrefixedAtomicStorage) ; ok && err == nil {
```

Comment:
@vsbogd , i dont like this type check very much here , any other thoughts  ? , other parts have shaped well